pkg/platform/infrastructure.go

```diff
@@ -0,0 +1,57 @@
+package platform
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/klog/v2"
+)
+
+var _ PlatformOwner = &InfrastrucutreOwner{}
```

Comment:
I'm not certain what the benefit of this abstraction is at the moment? What are we trying to achieve? Should we not just look up the `configv1.Infrastructure` at the beginning of the reconcile as we normally do in operators?

Reply:
https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/9#discussion_r565975485